infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogCompactor.java

```diff
@@ -144,18 +142,11 @@ private void compactSegment(Long ordinal) {
             CompactionFeedback compactionFeedback = compactStreamSegment(
                     inputStreamSegment, outputStreamSegment, inputGarbageSegment);
 
-            // We do not lock in the previous step to minimize the waiting time of hole fill writers.
-            try (AutoCloseableLock ignored =
-                         StreamLogSegment.getSegmentLock().acquireWriteLock(ordinal)) {
-                // Sync the delta updates during the initial compaction.
-                syncSegmentUpdate(inputStreamSegment, outputStreamSegment);
-                // Update global compaction mark before remapping to ensure following
-                // reads can return the most up-to-date compaction mark.
-                dataStore.updateGlobalCompactionMark(compactionFeedback.segmentCompactionMark);
-                dataStore.updateCompactedAddresses(ordinal, compactionFeedback.compactedAddresses);
-                logMetadata.pruneStreamSpace(compactionFeedback.addressesToPrune);
-                segmentManager.remapCompactedSegment(inputStreamSegment, outputStreamSegment);
-            }
+            // Update global compaction mark before remapping to ensure following
```

Comment:
The lock is removed. Is it because compact() is called only by one thread?

Reply:
Lock is removed because now we only compact addresses below committed tail, which means no one can write again to the segment.